etn-ccis/blui-react-native-component-library
# Update Icon component and docs

Fixes BLUI-4370.


#### Changes proposed in this Pull Request:
- Add/update icon component and docs to work with MD3


#### Screenshots / Screen Recording (if applicable)
![Simulator Screenshot - iPhone 14 - 2023-08-15 at 15 12 32](https://github.com/etn-ccis/blui-react-native-component-library/assets/13989985/a86d6b9a-81e2-4169-93f5-c66fbd46f177)



#### To Test:
- `cd md3`
- `yarn && cd ios && pod install && cd ..`
- `yarn ios` or `yarn android`
- observe various forms of icons rendered in channel values
- ensure all potential use cases are covered in the docs and examples

update: you can now test this in the showcase on branch `feature/blui-4370-icon-component`


#### Any specific feedback you are looking for?
